src/System.ServiceModel.Syndication/src/System/ServiceModel/Syndication/Atom10FeedFormatter.cs

```diff
@@ -1085,7 +1083,7 @@ private void ReadLink(XmlReader reader, SyndicationLink link, Uri baseUri)
             link.MediaType = mediaType;
             link.RelationshipType = relationship;
             link.Title = title;
-            link.Uri = (val != null) ? new Uri(val, UriKind.RelativeOrAbsolute) : null;
+            link.Uri = (val != null) ? UriParser(val, UriKind.RelativeOrAbsolute, Atom10Constants.LinkTag, ns) : null;
         }
```

Comment:
Do we want to do a similar thing with catching an exception and storing it until access for Uri's?

Reply:
The behavior would be more consistent if we make it that way. But it's not easy. We have more places using Uri then DateTime. And the result Uri would be used in different case, e.g. for property value, for constructor value, etc. I'd prefer to keep Uri as is.